z3/include/z3_term.h

```diff
@@ -29,60 +29,73 @@ namespace smt {
 // forward declaration
 class Z3Solver;
 
-class Z3TermIter : public TermIterBase
-{
- public:
-//   Z3TermIter(term_t t, uint32_t p) : term(t), pos(p){};
-  Z3TermIter(const Z3TermIter & it);
-  ~Z3TermIter(){};
-  Z3TermIter & operator=(const Z3TermIter & it);
-  void operator++() override;
-  const Term operator*() override;
-  TermIterBase * clone() const override;
-  bool operator==(const Z3TermIter & it);
-  bool operator!=(const Z3TermIter & it);
+class Z3TermIter: public TermIterBase {
+public:
+	Z3TermIter(expr t, uint32_t p) :
+			term(t), pos(p) {
+	}
+	;
+//	Z3TermIter(const Z3TermIter &it);
+	~Z3TermIter() {
+	}
+	;
+	Z3TermIter& operator=(const Z3TermIter &it);
+	void operator++() override;
+	const Term operator*() override;
+	TermIterBase* clone() const override;
+	bool operator==(const Z3TermIter &it);
+	bool operator!=(const Z3TermIter &it);
 
- protected:
-  bool equal(const TermIterBase & other) const override;
+protected:
+	bool equal(const TermIterBase &other) const override;
 
- private:
-//   term_t term;
-  uint32_t pos;
+private:
+	expr term;
+	uint32_t pos;
 };
 
-class Z3Term : public AbsTerm
-{
- public:
-  // assumes that term is not a function if flag is not passed
-//   Z3Term(term_t t) : term(t), is_function(false){};
-//   Z3Term(term_t t, bool is_fun) : term(t), is_function(is_fun){};
-  ~Z3Term(){};
-  std::size_t hash() const override;
-  bool compare(const Term & absterm) const override;
-  Op get_op() const override;
-  Sort get_sort() const override;
-  bool is_symbol() const override;
-  bool is_param() const override;
-  bool is_symbolic_const() const override;
-  bool is_value() const override;
-  virtual std::string to_string() override;
-  uint64_t to_int() const override;
-  /* Iterators for traversing the children */
-  TermIter begin() override;
-  TermIter end() override;
-  std::string print_value_as(SortKind sk) override;
+class Z3Term: public AbsTerm {
+public:
+	// Non-functions
+   Z3Term(expr t, context &c) : term(t), is_function(false), z_func(c) {
+	   ctx = &c;
+   };
+   // Functions
+   Z3Term(func_decl zfunc, context &c) : term(c), is_function(true), z_func(zfunc) {
+	   ctx = &c;
+   };
+	~Z3Term() {
```

Comment:
Looks like there's some formatting weirdness here. Maybe from the merge conflicts? Although that isn't supposed to happen. As we discussed earlier though, you can just run clang-format later, so nothing to be done now.